src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,454 @@
+    React.useEffect(() => {
+      setCurrentValues(values);
+    }, [values]);
+
+    const [minValueLabel, setMinValueLabel] = React.useState(minLabel);
+    React.useEffect(() => {
+      setMinValueLabel(minLabel ?? min.toString());
+    }, [minLabel, min]);
+
+    const [maxValueLabel, setMaxValueLabel] = React.useState(maxLabel);
+    React.useEffect(() => {
+      setMaxValueLabel(maxLabel ?? max.toString());
+    }, [maxLabel, max]);
+
+    const [trackDisplay, setTrackDisplay] = React.useState<TrackDisplayMode>(
+      () => getDefaultTrackDisplay(trackDisplayMode, currentValues),
+    );
+    React.useEffect(() => {
+      setTrackDisplay(getDefaultTrackDisplay(trackDisplayMode, currentValues));
+    }, [trackDisplayMode, currentValues]);
+
+    useTheme();
+    const containerRef = React.useRef<HTMLDivElement>(null);
+
+    React.useEffect(() => {
+      if (containerRef.current && setFocus) {
+        const firstThumb = containerRef.current.querySelector(
+          '.iui-slider-thumb',
+        );
+        firstThumb && (firstThumb as HTMLElement).focus();
+      }
+    }, [setFocus]);
+
+    const fractionDigits = React.useMemo(() => {
+      const stepString = step.toString();
+      const decimalIndex = stepString.indexOf('.');
+      return stepString.length - (decimalIndex + 1);
+    }, [step]);
+
+    const generateTooltip = React.useCallback(
+      (val: number): React.ReactNode => {
+        return tooltipRenderer
+          ? tooltipRenderer(val, step)
+          : formatNumberValue(val, step, fractionDigits);
+      },
+      [fractionDigits, step, tooltipRenderer],
+    );
+
+    const getAllowableThumbRange = React.useCallback(
+      (index: number) => {
+        if (thumbMode === 'inhibit-crossing') {
+          const minVal = index === 0 ? min : currentValues[index - 1] + step;
+          const maxVal =
+            index < currentValues.length - 1
+              ? currentValues[index + 1] - step
+              : max;
+          return [minVal, maxVal];
+        }
+        return [min, max];
+      },
+      [max, min, step, thumbMode, currentValues],
+    );
+
+    const [activeThumbIndex, setActiveThumbIndex] = React.useState<
+      number | undefined
+    >(undefined);
+
+    const updateThumbValue = React.useCallback(
+      (
+        event: PointerEvent,
+        callback?: (values: ReadonlyArray<number>) => void,
+      ) => {
+        if (containerRef.current && undefined !== activeThumbIndex) {
+          const percent = getPercentageOfRectangle(
+            containerRef.current.getBoundingClientRect(),
+            event.pageX,
+          );
+          let pointerValue = min + (max - min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step, min);
+          const [minVal, maxVal] = getAllowableThumbRange(activeThumbIndex);
+          pointerValue = getBoundedValue(pointerValue, minVal, maxVal);
+          if (pointerValue === currentValues[activeThumbIndex]) {
```

Comment:
This part of the code is same as in `onThumbValueChanged`.. Wonder if we could simplify this, but not sure I want another function with useCallback :D It is already a lot of those, where 1 prop change could cause some of them to update refs...